src/attributes/codegen.md

```diff
@@ -66,20 +65,42 @@ It is [undefined behavior] to call a function that is compiled with a feature
 that is not supported on the current platform the code is running on, *except*
 if the platform explicitly documents this to be safe.
 
-Functions marked with `target_feature` are not inlined into a context that
-does not support the given features. The `#[inline(always)]` attribute may not
-be used with a `target_feature` attribute.
+For this reason, a function marked with `target_feature` is unsafe, except in
+a context that supports the given features. For example:
+
+```rust
+# #[target_feature(enable = "avx2")]
```

Comment:
This will similarly need a cfg to restrict it to platforms that support avx2.  You can do something similar to #1334.